app/models/vim_performance_state.rb

```diff
@@ -197,8 +205,15 @@ def self.capture_vm_disk_storage(obj, field)
   end
 
   def self.capture_cpu_total_cores(obj)
-    return unless obj.kind_of?(VmOrTemplate)
-    obj.hardware.try(:cpu_total_cores)
+    if obj.respond_to?(:hardware)
+      hardware = obj.hardware
+    elsif obj.respond_to?(:container_node)
+      hardware = obj.container_node.hardware
+    else
+      return nil
+    end
```

Comment:
This feels like a good candidate for refactoring later ... maybe having provider specific `performance_state` models...  Not for this PR, though.